homeassistant/components/nam/sensor.py

```diff
@@ -381,19 +422,8 @@ def available(self) -> bool:
             and getattr(self.coordinator.data, self.entity_description.key) is not None
         )
 
-    @property
-    def options(self) -> list[str] | None:
-        """If the entity description provides a mapping, use that."""
-        if self.entity_description.mapping:
-            return list(self.entity_description.mapping.values())
-        return super().options
-
-
-class NAMSensorUptime(NAMSensor):
-    """Define an Nettigo Air Monitor uptime sensor."""
-
-    @property
-    def native_value(self) -> datetime:
-        """Return the state."""
-        uptime_sec = getattr(self.coordinator.data, self.entity_description.key)
-        return utcnow() - timedelta(seconds=uptime_sec)
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_native_value = self.entity_description.value(self.coordinator.data)
+        self.async_write_ha_state()
```

Comment:
Just wondering here... what is the benefit of a callback over native_value?

```python
    @property
    def native_value(self) -> StateType | datetime:
        """Return the state."""
        return self.entity_description.value(self.coordinator.data)
```

Reply:
In most cases, reviewers prefer when `_attr_*` attributes are used instead of properties. I think the code looks cleaner when we use `_attr_* and a callback method to update the state/extra state attributes.

Reply:
I think the main exception to the `_attr_*` rule is when the property is based on coordinator data...
I think in this case the `native_value` is cleaner/easier to undestand.

Maybe a second opinion here?

Reply:
Sure, we need a second pair of eyes 😀

Reply:
In this case, the property method should be used instead.

There is no processing of data, and all data is already available. It can be picked up from the coordinator directly.

../Frenck

Reply:
Thanks guys. Property reverted.